src/ccmain/tessedit.cpp

```diff
@@ -81,14 +82,11 @@ bool Tesseract::init_tesseract_lang_data(const std::string &arg0,
                                          bool set_only_non_debug_params, TessdataManager *mgr) {
   // Set the language data path prefix
   lang = !language.empty() ? language : "eng";
-  language_data_path_prefix = datadir;
-  language_data_path_prefix += lang;
-  language_data_path_prefix += ".";
+  std::filesystem::path tessdata_path = datadir / (lang + "." + kTrainedDataSuffix);
 
   // Initialize TessdataManager.
-  std::string tessdata_path = language_data_path_prefix + kTrainedDataSuffix;
-  if (!mgr->is_loaded() && !mgr->Init(tessdata_path.c_str())) {
-    tprintf("Error opening data file %s\n", tessdata_path.c_str());
+  if (!mgr->is_loaded() && !mgr->Init(tessdata_path.string().c_str())) {
+    tprintf("Error opening data file %s\n", tessdata_path.string().c_str());
```

Comment:
Here the new stream output would not need c_str().

Reply:
Fuzzer does not like it:
```src/ccmain/tessedit.cpp:89:45: error: cannot pass non-trivial object of type 'std::string' (aka 'basic_string<char>') to variadic function; expected type from format string was 'char *' [-Wnon-pod-varargs]
   89 |     tprintf("Error opening data file %s\n", tessdata_path.string());
      |                                      ~~     ^~~~~~~~~~~~~~~~~~~~~~
src/ccmain/tessedit.cpp:89:45: note: did you mean to call the c_str() method?
```

Reply:
Yes, tprintf had to be replaced by tesserr. I fixed this now.